### PR TITLE
SDIT-2691 Add response fields query parameter to /prisoner-numbers endpoint

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerSearchResource.kt [hunk 1]
```diff
@@ -122,19 +127,11 @@ class PrisonerSearchResource(
...
     )
     responseFields: List<String>? = null,
     @ParameterObject @PageableDefault
     pageable: Pageable,
-  ): Page<Prisoner> {
-    responseFields?.run {
-      responseFieldsValidator.findMissing(responseFields)
-        .takeIf { it.isNotEmpty() }
-        ?.run { throw BadRequestException("Invalid response fields requested: $this") }
-    }
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerSearchResource.kt:136]: Pushed the throw into the validator and moved the validator into the service layer.

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerSearchByPrisonerNumbersResourceTest.kt [hunk 2]
```diff
@@ -88,4 +89,47 @@ class PrisonerSearchByPrisonerNumbersResourceTest : AbstractSearchIntegrationTes
       .exchange()
       .expectStatus().isForbidden
   }
+
+  @Test
+  fun `returns only requested response fields`() {
+    webTestClient.post()
+      .uri {
+        it.path("/prisoner-search/prisoner-numbers")
+          .queryParam("responseFields", "prisonerNumber")
```
Comment (petergphillips) [inline hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerSearchByPrisonerNumbersResourceTest.kt:98]: would also be good to test for an empty list being sent as `responseFields` - as opposed to query param not being present

**Review by petergphillips — APPROVED** [review]
LGTM

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerSearchByPrisonerNumbersResourceTest.kt [hunk 3]
```diff
@@ -88,4 +89,47 @@ class PrisonerSearchByPrisonerNumbersResourceTest : AbstractSearchIntegrationTes
       .exchange()
       .expectStatus().isForbidden
   }
+
+  @Test
+  fun `returns only requested response fields`() {
```
[the remark this answers is not in the document]
Comment (mikehalmamoj) [reply hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerSearchByPrisonerNumbersResourceTest.kt:94]: Lies - the "comprehensive tests" are in PrisonerResponseFieldsTest

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerSearchByPrisonerNumbersResourceTest.kt (hunk 2 above)
Comment (mikehalmamoj) [reply hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/PrisonerSearchByPrisonerNumbersResourceTest.kt:98]: > would also be good to test for an empty list being sent as `responseFields` - as opposed to query param not being present

This one is actually tested in `PrisonerResponseFieldsTest` / `should return all fields if empty list of fields requested`.
